Let update-rc.d set changed flag properly

madduck opened the pull request:
When update-rc.d is used to enable/disable service, the changed flag was
always true (see #2189). This commit fixes that.

Signed-off-by: martin f. krafft madduck@madduck.net

madduck committed:
Let update-rc.d set changed flag properly

When update-rc.d is used to enable/disable service, the changed flag was
always true (see #2189). This commit fixes that.

Signed-off-by: martin f. krafft <madduck@madduck.net>

mpdehaan commented:
Thanks...

The single line if/else statement is not Python 2.4 compatible and needs to be broken out.

madduck commented:
Good catch!

madduck committed:
Expand usage of Python 2.6 if statement

Syntax like "'foo' if bar else 'baz'" is not supported by all Python
versions targetted by Ansible. Hence we break it up.

Signed-off-by: martin f. krafft <madduck@madduck.net>

mpdehaan commented:
merged, thanks!

artnez commented:
Looks like the fixed regressed. Disabling a service will always issue a change event.
